src/lib/PageTracker.js

```diff
@@ -0,0 +1,347 @@
+
+    /**
+     * Call the resetIdleTimer if the user is active
+     *
+     * @private
+     * @return {Function} Throttled activeUser handler
+     */
+    activeUserHandler() {
+        return throttle(() => {
+            this.resetIdleTimer();
+        }, ACTIVE_USER_THROTTLE);
+    }
+
+    /**
+     * Stops the content insights interval if the user goes inactive
+     *
+     * @private
+     * @return {void}
+     */
+    inactiveUserHandler() {
+        this.stopTracking();
+    }
+
+    /**
+     * Binds DOM listeners to a container to manage if the user goes active or inactive.
+     *
+     * @private
+     * @param {HTMLElement} container - File Preview container
+     * @return {void}
+     */
+    bindIdleUserEvents(container) {
+        if (container) {
+            this.container = container;
+            this.container.addEventListener('mousemove', this.throttledActiveUserHandler);
+            this.container.addEventListener('mouseenter', this.throttledActiveUserHandler);
+            this.container.addEventListener('mouseleave', this.inactiveUserHandler);
+            this.container.addEventListener('focusin', this.throttledActiveUserHandler);
+            this.container.addEventListener('focusout', this.inactiveUserHandler);
+            this.container.addEventListener('click', this.throttledActiveUserHandler);
+            this.container.addEventListener('wheel', this.throttledActiveUserHandler);
+            this.container.addEventListener('keydown', this.throttledActiveUserHandler);
+            this.container.addEventListener('touchstart', this.throttledActiveUserHandler);
+            this.container.addEventListener('touchmove', this.throttledActiveUserHandler);
+        }
+    }
+
+    /**
+     * Unbinds DOM listeners from the container that manage if the user goes active or inactive.
+     *
+     * @private
+     * @return {void}
+     */
+    unbindIdleUserEvents() {
+        if (this.container) {
+            this.container.removeEventListener('mousemove', this.throttledActiveUserHandler);
+            this.container.removeEventListener('mouseenter', this.throttledActiveUserHandler);
+            this.container.removeEventListener('mouseleave', this.inactiveUserHandler);
+            this.container.removeEventListener('focusin', this.throttledActiveUserHandler);
+            this.container.removeEventListener('focusout', this.inactiveUserHandler);
+            this.container.removeEventListener('click', this.throttledActiveUserHandler);
+            this.container.removeEventListener('wheel', this.throttledActiveUserHandler);
+            this.container.removeEventListener('keydown', this.throttledActiveUserHandler);
+            this.container.removeEventListener('touchstart', this.throttledActiveUserHandler);
+            this.container.removeEventListener('touchmove', this.throttledActiveUserHandler);
+        }
+    }
+
+    /**
+     *  Construct and send the payload object to send to the BFF endpoint.
```

Comment:
Since this is a OSS library, we don't like to expose what our proprietary endpoints are, could this follow the existing pattern of emitting an event that is consumed by a parent application?